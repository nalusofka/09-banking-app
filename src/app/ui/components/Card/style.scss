@use '../../styles/variables' as var;

.card {
  background: var.$color-light-gray2;
  border-radius: 8px;
  box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
  display: grid;
  grid-template-columns: 1fr auto;
  padding: 20px;
  text-decoration: inherit;
  transition: transform 0.3s ease;

  &-content {
    display: flex;
    flex-direction: column;
  }

  &-title {
    margin-bottom: 10px;
    color: var.$color-dark-blue;
    font-size: 1.5rem;
  }

  &-description {
    color: var.$color-gray;
    font-size: 1rem;
  }

  &-link {
    align-items: center;
    color: var.$color-light-gray;
    display: flex;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }
  }
}
