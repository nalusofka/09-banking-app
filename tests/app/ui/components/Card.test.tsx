import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Card from "../../../../src/app/ui/components/Card";

describe("Componente Card", () => {
  const mockProps = {
    title: "Título de Prueba",
    description: "Descripción de prueba para el componente Card.",
    to: "/ruta-de-prueba",
  };

  it("renderiza correctamente el título y la descripción", () => {
    const { getByText } = render(
      <Router>
        <Card {...mockProps} />
      </Router>
    );

    expect(getByText(mockProps.title)).toBeInTheDocument();
    expect(getByText(mockProps.description)).toBeInTheDocument();
  });

  it("contiene el enlace correcto", () => {
    const { container } = render(
      <Router>
        <Card {...mockProps} />
      </Router>
    );

    const links = container.querySelectorAll("a");

    links.forEach((link) => {
      expect(link).toHaveAttribute("href", mockProps.to);
    });
  });

  it("renderiza el ícono correctamente", () => {
    const { container } = render(
      <Router>
        <Card {...mockProps} />
      </Router>
    );

    const icon = container.querySelector("svg");
    expect(icon).toBeInTheDocument();
  });

  it("aplica las clases CSS correctamente", () => {
    const { container } = render(
      <Router>
        <Card {...mockProps} />
      </Router>
    );

    const card = container.querySelector(".card");
    const cardTitle = container.querySelector(".card-title");
    const cardDescription = container.querySelector(".card-description");
    const cardLink = container.querySelector(".card-link");

    expect(card).toHaveClass("card");
    expect(cardTitle).toHaveClass("card-title");
    expect(cardDescription).toHaveClass("card-description");
    expect(cardLink).toHaveClass("card-link");
  });

  it("genera un snapshot válido", () => {
    const { asFragment } = render(
      <Router>
        <Card {...mockProps} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
