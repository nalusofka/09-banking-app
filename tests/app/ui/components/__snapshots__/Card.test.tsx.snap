// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Card > genera un snapshot válido 1`] = `
<DocumentFragment>
  <a
    class="card"
    href="/ruta-de-prueba"
  >
    <div
      class="card-content"
    >
      <h3
        class="card-title title-secondary"
        data-testid="title-component"
      >
        Título de Prueba
      </h3>
      <p
        class="card-description"
        data-testid="body-component"
      >
        Descripción de prueba para el componente Card.
      </p>
    </div>
  </a>
  <a
    class="card-link"
    href="/ruta-de-prueba"
  >
    <svg
      fill="currentColor"
      height="30"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
      />
    </svg>
  </a>
</DocumentFragment>
`;
